File: js/add-two-numbers.js
function ListNode(val) {
  this.val = val
  this.next = null
}
/**
 * @param {ListNode} l1 头节点
 * @param {ListNode} l2 头节点
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
  const head = new ListNode(null)
  let cur = head
  let carry = 0
  
  while (l1 || l2 || carry) {
      let sum = carry
      if (l1) {
          sum += l1.val
          l1 = l1.next
      }
      if (l2) {
          sum += l2.val
          l2 = l2.next
      }
      
      carry = sum >= 10 ? 1 : 0
      cur.next = new ListNode(sum % 10)
      cur = cur.next
  }
  
  return head.next
};

File: py/add-two-numbers.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        Head = ListNode(0)
        curr = Head
        carry = 0

        while l1 or l2:
            global carry
            val1 = l1.val if l1 else 0
            val2 = l2.val if l2 else 0
            sum = val1 + val2 + carry

            if (sum > 9):
                curr.next = ListNode(sum - 10)
                carry = 1
            else:
                curr.next = ListNode(sum)
                carry = 0
            curr = curr.next

            if l1:
                l1 = l1.next
            if l2:
                l2 = l2.next

        if (carry):
            curr.next = ListNode(1)

        return Head.next
